### PR TITLE
Block more resource loading to parent

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Block service resource loading to parent.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
`ServiceLoader` tries to find resources to understand what classes to load, and for the following scenario, it will fail with a mysteries error message.

* `com.example.Foo` exists both in parent and child classpath, in two different jars
* `com.example.FooImpl` only exists in parent classpath
* Task code (under child classpath) uses `ServiceLoader` to discover implementation of `com.example.Foo`, using `ChildFirstClassLoader`
* `ServiceLoader` uses `ChildFirstClassLoader` to find resources from both child and parent classpath, and in this case `com.example.FooImpl` is discovered
* `ServiceLoader` verifies whether `com.example.FooImpl` (loaded by parent class loader, and its super class `com.example.Foo` is also loaded by parent class loader; remember there are two copies of `com.example.Foo`?) can be assigned to `com.example.Foo` (loaded by child class loader)
* The verification fails because there are two copies of `com.example.Foo` loaded by different class loaders, with an error message like "java.util.ServiceConfigurationError: com.example.Foo: com.example.FooImpl not a subtype"

## Tracking Issue
Closes https://github.com/flyteorg/flyte/issues/3103

## Follow-up issue
_NA_

### CODE REVIEW

File: integration-tests/src/test/java/org/flyte/JavaExamplesIT.java [hunk 1]
```diff
@@ -63,6 +64,7 @@ public void testFibonacciWorkflow() {
   }
 
   @Test
+  @Disabled
```
Comment (honnix) [inline integration-tests/src/test/java/org/flyte/JavaExamplesIT.java:67]: For some reason this always times out.

**Review by narape — APPROVED** [review]
